lib/game.rb

```diff
@@ -0,0 +1,129 @@
+require 'pry'
+class Game
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [[0,1,2], [3,4,5], [6,7,8], [0,3,6], [1,4,7], [2,5,8], [0,4,8], [2,4,6]]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+  
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+  
+  def won?
+    WIN_COMBINATIONS.detect do |combo|
+      if board.taken?(combo[0] + 1) && board.cells[combo[0]] == board.cells[combo[1]] && board.cells[combo[1]] == board.cells[combo[2]]
+        return combo
+      end
+    end
+  end 
+  
+  def draw?
+    board.full? && !won?
+  end 
+  
+  def over?
+    won? || draw?
+  end
+  
+  def winner
+    if won?
+      combo = won?
+      board.cells[combo[0]]
+    end
+  end
+    
+  def turn
+    num = current_player.move(board)
+    while !board.valid_move?(num)
+      num = current_player.move(board)
+    end
+    board.update(num, current_player)
+    board.display
+  end
+    
+  def play 
+    turn until over?
+    board.display
+    puts winner ? "Congratulations #{winner}!" : "Cat's Game!"
+  end
+  
+
+  def start
```

Comment:
Great bin file, short length is ideal. However, we would prefer to see the CLI logic separated from the game logic in its own class (e.g. TicTacToeCLI). Splitting the interface from the implementation of the game allows for multiple interfaces (e.g. a web app) without requiring changes to the game code.